beancount/parser/parser.py

```diff
@@ -214,9 +218,12 @@ def parse_string(string, report_filename=None, **kw):
     """
     if kw.pop('dedent', None):
         string = textwrap.dedent(string)
-    builder = grammar.Builder(report_filename or '<string>')
-    _parser.parse_string(string, builder, report_filename=report_filename, **kw)
-    return builder.finalize()
+    if isinstance(string, str):
+        string = string.encode('utf8')
+    if report_filename is None:
+        report_filename = '<string>'
+    file = io.BytesIO(string)
```

Comment:
Does this make a copy?

Reply:
Constructing the `io.BytesIO` object (line 225) does not. Encoding the docstring in utf8 (line 222) does, unless Python is very smart.